motor_control/throttle/button_control.ino

```diff
@@ -8,22 +8,27 @@ ButtonThrottle::ButtonThrottle(const int &pin, int timeToFull) : Throttle(pin, t
 
 int ButtonThrottle::mappingFunction(int time_pressed, m_map_type mapType)
 {
-  if (mapType == LINEAR)
+  if(m_isPressed == true)
   {
-    return std::min((double)time_pressed / (double)m_timeToFull, 1.0) * 100;
+if (mapType == LINEAR)
+  {
+    m_speedOfCar = std::min((double)time_pressed / (double)m_timeToFull, 1.0) * m_maxValue;
```

Comment:
If you want to do floating point division, you only need one variable to be a double. If we changed the type of _time_pressed_ to _double_ in the function prototype we could avoid type casting here and on line 22, which would make the code a bit cleaner without any loss in precision.

Reply:
This might also help with lines 35-37 in replacing the variable _temp_time_pressed_